ci_test/unit_tests/test_unit_subgrid_cross_grid_sum_opt.py

```diff
@@ -0,0 +1,183 @@
+import functools

```

Comment:
General note: should the test file names change to subgraph?

Reply:
We can. Let's discuss it in the meeting.